### PR TITLE
Datasources for ipsec and ike policy and collection

### PR DESCRIPTION
**Pull request by deepaksibm**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

**Comment by deepaksibm** [comment]
```
$ make testacc TEST=./ibm TESTARGS='-run=TestAccIBMIsIpsecPolicyDataSourceBasic'
=== RUN   TestAccIBMIsIpsecPolicyDataSourceBasic
--- PASS: TestAccIBMIsIpsecPolicyDataSourceBasic (102.65s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm 105.239s

$ make testacc TEST=./ibm TESTARGS='-run=TestAccIBMIsIpsecPoliciesDataSourceBasic'
=== RUN   TestAccIBMIsIpsecPoliciesDataSourceBasic
--- PASS: TestAccIBMIsIpsecPoliciesDataSourceBasic (41.20s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm 43.747s

$ make testacc TEST=./ibm TESTARGS='-run=TestAccIBMIsIkePolicyDataSourceBasic'
=== RUN   TestAccIBMIsIkePolicyDataSourceBasic
--- PASS: TestAccIBMIsIkePolicyDataSourceBasic (94.32s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm 97.921s

$ make testacc TEST=./ibm TESTARGS='-run=TestAccIBMIsIkePoliciesDataSourceBasic'
=== RUN   TestAccIBMIsIkePoliciesDataSourceBasic
--- PASS: TestAccIBMIsIkePoliciesDataSourceBasic (38.61s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm 41.431s

```

File: ibm/data_source_ibm_is_ike_policies_test.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+					resource.TestCheckResourceAttrSet("data.ibm_is_ike_policies.is_ike_policies", "ike_policies.#"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckIBMIsIkePoliciesDataSourceConfigBasic() string {
+	return fmt.Sprintf(`
+		data "ibm_is_ike_policies" "is_ike_policies" {
```
Comment (hkantare) [inline ibm/data_source_ibm_is_ike_policies_test.go:30]: Can you first ceate a ike policy and then call the datasource to list

File: ibm/data_source_ibm_is_ike_policy.go [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+		ReadContext: dataSourceIBMIsIkePolicyRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ExactlyOneOf: []string{"name", "ike_policy_id"},
+				Description:  "The IKE policy name.",
+			},
+			"ike_policy_id": {
```
Comment (hkantare) [inline ibm/data_source_ibm_is_ike_policy.go:28]: Can you please just rename to ike_policy just to be line with all other resources

File: ibm/data_source_ibm_is_ike_policy.go [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+				return diag.FromErr(fmt.Errorf("Error Fetching IKE Policies %s\n%s", err, response))
+			}
+			start = GetNext(ikePolicies.Next)
+			allrecs = append(allrecs, ikePolicies.IkePolicies...)
+			if start == "" {
+				break
+			}
+		}
+		for _, ikePolicyItem := range allrecs {
+			if *ikePolicyItem.Name == name {
```
Comment (hkantare) [inline ibm/data_source_ibm_is_ike_policy.go:178]: if there is no ike policy with given name are we throwing nay error

File: ibm/data_source_ibm_is_ipsec_policies_test.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+					resource.TestCheckResourceAttrSet("data.ibm_is_ipsec_policies.is_ipsec_policies", "ipsec_policies.#"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckIBMIsIpsecPoliciesDataSourceConfigBasic() string {
+	return fmt.Sprintf(`
+		data "ibm_is_ipsec_policies" "is_ipsec_policies" {
```
Comment (hkantare) [inline ibm/data_source_ibm_is_ipsec_policies_test.go:30]: same create on ipsec policy and then call datasource

File: ibm/data_source_ibm_is_ipsec_policy.go [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+		ReadContext: dataSourceIBMIsIpsecPolicyRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ExactlyOneOf: []string{"name", "ipsec_policy_id"},
+				Description:  "The IPsec policy name.",
+			},
+			"ipsec_policy_id": {
```
Comment (hkantare) [inline ibm/data_source_ibm_is_ipsec_policy.go:28]: rename to ipsec_policy

File: website/docs/d/is_ipsec_policy.html.markdown [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+## Example Usage
+
+```hcl
+resource "ibm_is_ipsec_policy" "example" {
+  name                     = "my-ipsec-policy"
+  authentication_algorithm = "md5"
+  encryption_algorithm     = "triple_des"
+  pfs                      = "disabled"
+}
+
```
Comment (hkantare) [inline website/docs/d/is_ipsec_policy.html.markdown:22]: format this example

File: website/docs/d/is_ike_policy.html.markdown [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+subcategory: "VPC infrastructure"
+layout: "ibm"
+page_title: "IBM : ibm_is_ike_policy"
+description: |-
+  Get information about IKEPolicy
+---
+
+# ibm_is_ike_policy
+
+Provides a read-only data source for IKEPolicy. You can then reference the fields of the data source in other resources within the same configuration using interpolation syntax. For more information, about managing IBM Cloud VPN Gateway and IKE policy , see [about site-to-site VPN gateways](https://cloud.ibm.com/docs/vpc?topic=vpc-using-vpn&interface=ui#policy-negotiation).
```
Comment (hkantare) [inline website/docs/d/is_ike_policy.html.markdown:11]: Format example

**Comment by deepaksibm** [comment]
```
$ make testacc TEST=./ibm/service/vpc TESTARGS='-run=TestAccIBMIsIpsecPoliciesDataSourceBasic'
=== RUN   TestAccIBMIsIpsecPoliciesDataSourceBasic
--- PASS: TestAccIBMIsIpsecPoliciesDataSourceBasic (45.76s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     47.369s

$ make testacc TEST=./ibm/service/vpc TESTARGS='-run=TestAccIBMIsIpsecPolicyDataSourceBasic'
=== RUN   TestAccIBMIsIpsecPolicyDataSourceBasic
--- PASS: TestAccIBMIsIpsecPolicyDataSourceBasic (76.01s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     76.882s

$ make testacc TEST=./ibm/service/vpc TESTARGS='-run=TestAccIBMIsIkePoliciesDataSourceBasic'
=== RUN   TestAccIBMIsIkePoliciesDataSourceBasic
--- PASS: TestAccIBMIsIkePoliciesDataSourceBasic (43.02s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     44.104s

$ make testacc TEST=./ibm/service/vpc TESTARGS='-run=TestAccIBMIsIkePolicyDataSourceBasic'
=== RUN   TestAccIBMIsIkePolicyDataSourceBasic
--- PASS: TestAccIBMIsIkePolicyDataSourceBasic (86.14s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     87.040s
```

**Comment by deepaksibm** [comment]
Conflicts resolved

**Comment by hkantare** [comment]
Can you please fix the conflict

**Comment by deepaksibm** [comment]
> Can you please fix the conflict

conflict fixed , thanks

**Pull request merged**